### PR TITLE
ESLint: Add inclusive language rule #43600

### LINKED ISSUE #43600: Add ESLint rule to error on problematic language
In order to avoid introducing new [instances of problematic language](https://github.com/Automattic/wp-calypso/issues/43395).

To accomplish this we could use the [`id-blacklist`](https://eslint.org/docs/rules/id-blacklist) rule. While as of today it is woefully (and in this case quite ironically) named, the ESLint project is undergoing the same process as us in removing this kind of problematic language: https://github.com/eslint/eslint/issues/13407

We could either (a) go ahead and add it using the current `id-blacklist` named rule _or_ (b) wait until ESLint have updated it, upgrade our dependency and then use the newly renamed rule. Either way, we should keep our eyes on https://github.com/eslint/eslint/pull/13408, once it is merged and a new ESLint release is cut we can do the upgrade.

As far as the list of terms to add to the rule configuration, I think we could start with the following:
1. "whitelist"
2. "blacklist"
3. "master"
4. "slave"

### PR DESCRIPTION
**Pull request by sarayourfriend**
#### Changes proposed in this Pull Request

* Add [`eslint-plugin-inclusive-language`](https://github.com/muenzpraeger/eslint-plugin-inclusive-language) to our global `.eslintrc`. This will prevent non-inclusive language from being reintroduced into our repository.

This is a major version bump and will require a package re-release after merging.

#### Testing instructions

* `yarn install` and then attempt to add one of the non-inclusive terms from the plugin: https://github.com/muenzpraeger/eslint-plugin-inclusive-language/blob/primary/lib/config/inclusive-words.json

#### In action

Captura de Tela 2020-06-25 às 08 48 56

Fixes #43600

### CODE REVIEW

**Comment by jsnajdr** [comment]
I'm curious how many issues do we get reported in our codebase, and how many of them are identifiers that are outside Calypso control. I see a setting named `jetpack_protect_global_whitelist`, and a `whitelist` parameter passed to the `webpack-node-externals` plugin.

File: .eslintrc.js
Comment (jsnajdr) [inline .eslintrc.js:181]: A better place for this rule definition would be the `eslint-config-wpcalypso` package. It's reusable in other projects and the `/.eslintrc.js` file contains only overrides that are specific to the Calypso app in particular.

**Review by jsnajdr — APPROVED** [review]
Looks good. Let's move the rule to `eslint-config-wpcalypso` before merging, but it's not a strict blocker.

**Comment by saramarcondes** [comment]
> I'm curious how many issues do we get reported in our codebase

Once I'm done merging the various PRs I have open (others are participating as well) as part of #43395 we should only get the jetpack error codes you pointed out (as well as a couple different places we cannot change, things stored in post meta, site settings names, etc). One thing I could do with this PR is wait until I've merged everything and then add an eslint disable pragma for the places I know/suspect won't be able to be changed. At least then there is an explanation and explicit acknowledgement that the language is not good, but that we cannot currently do anything about it.

Once #43395 is addressed it should only be a handful of errors that we can disable.

**Comment by saramarcondes** [comment]
@griffbrad Adding you as a reviewer hoping you could look over and make sure I'm taking care of everything that is needed when adding a new dependency to this package. I'm following the model of this PR: https://github.com/Automattic/wp-calypso/pull/37563

**Review by lynnbgriffith — APPROVED** [review]
👍🏻 Yes.  Makes sense to me.  Added the dependency in Calypso's root package.json and made it a peerDependency of the package itself.

**Pull request merged**
